tests/pna/pixeldataset/test_io.py

```diff
@@ -182,7 +182,8 @@ def test_sample_names(self, pxl_view):
         assert pxl_view.sample_names() == ["test_sample"]
 
     def test_read_adata_from_sample(self, pxl_view: PixelDataViewer, adata_data):
-        res = pxl_view.read_adata_from_sample("test_sample")
+        with pxl_view as connection:
+            res = pxl_view.read_adata_from_sample(connection, "test_sample")
```

Comment:
I'm not entirely satisfied with how the UI looks like here, i.e that this relies on the user not mixing `pxl_view` and `connection` from different files. More specifically I would prefer if `connection` was encapsulated 

Maybe it's possible to achieve something similar to how regular files work? With a context manager it would look something like:
```python
with PxlDataViewer.from_files(["/path/to/file"]) as pxl_view:
    pxl_view.read_adata_from_sample("test_sample")
```